### PR TITLE
25 conditional error

### PR DESCRIPTION
**Pull request by nicomitchell**
* Fixed the `if item == 0 or -1` error from #25  :grimacing:

### CODE REVIEW

**Comment by rgabbard** [comment]
@nicomitchell : We also need to add a test which would have failed before this fix

File: tests/test_immutableset.py [hunk 1]
```diff
@@ -200,6 +200,8 @@ def test_singleton_index(self):
 
         self.assertEqual(1, s[0])
         self.assertEqual(1, s[-1])
+        with self.assertRaises(IndexError):
+            s[2]
```
Comment (ConstantineLignos) [inline tests/test_immutableset.py]: Can you add another test for a negative out of bounds, like -2?

File: tests/test_immutableset.py [hunk 2]
```diff
@@ -200,6 +200,11 @@ def test_singleton_index(self):
 
         self.assertEqual(1, s[0])
         self.assertEqual(1, s[-1])
+        with self.assertRaises(IndexError):
```
Comment (ConstantineLignos) [inline tests/test_immutableset.py]: You need a `with` block for each line. This just tests that at least one of these raises an exception.

Comment (nicomitchell) [reply tests/test_immutableset.py]: good catch

**Comment by rgabbard** [comment]
@ConstantineLignos : is this clear to merge?

**Comment by ConstantineLignos** [comment]
Looks good to me.
